### ISSUE TITLE
Can't new up arrays in Select

### ISSUE THREAD

**Issue by jcansdale** [grokys/Octokit.GraphQL#14]
For example, the following:

`.Nodes.Select(r => string.Join(";", new string[] { r.Owner.Login, r.Name }))`

Blows up with:

```
	System.InvalidOperationException: An expression of type 'Newtonsoft.Json.Linq.JToken' cannot be used to initialize an array of type 'System.String'
	at System.Linq.Expressions.Expression.NewArrayInit(Type type, IEnumerable`1 initializers)
	at System.Linq.Expressions.NewArrayExpression.Update(IEnumerable`1 expressions)
	at System.Linq.Expressions.ExpressionVisitor.VisitNewArray(NewArrayExpression node)
	at System.Linq.Expressions.NewArrayExpression.Accept(ExpressionVisitor visitor)
	at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
	at Octokit.GraphQL.Core.Builders.QueryBuilder.<VisitMethodArguments>d__15.MoveNext()
```

**Comment by StanleyGoldman** [comment]
Yea, the more I learn about GraphQL, I do not believe this is a bug.
Creating an array is not a valid GraphQL operation

**Comment by StanleyGoldman** [comment]
I also am not sure how to wrap it up with a better exception

**Comment by grokys** [comment]
This is actually a bug - creating the array is a client-side feature, it's just that the `QueryBuilder` didn't handle `NewArray` expressions. Fixed in #92.
